Cabal/Distribution/Simple/GHC/IPIConvert.hs

```diff
@@ -9,31 +9,31 @@
 --
 -- Helper functions for 'Distribution.Simple.GHC.IPI642'.
 module Distribution.Simple.GHC.IPIConvert (
-    PackageIdentifier, convertPackageId,
+    MungedId, convertMungedId,
     License, convertLicense,
     convertModuleName
   ) where
 
 import Prelude ()
 import Distribution.Compat.Prelude
 
-import qualified Distribution.Types.PackageId as Current
-import qualified Distribution.Types.PackageName as Current
+import qualified Distribution.Types.MungedId as Current
+import qualified Distribution.Types.MungedName as Current
 import qualified Distribution.License as Current
 
 import Distribution.Version
 import Distribution.ModuleName
 import Distribution.Text
 
-data PackageIdentifier = PackageIdentifier {
-    pkgName    :: String,
-    pkgVersion :: Version
+data MungedId = MungedId {
```

Comment:
This isn't right. The reason why PackageIdentifier is redefined here is because the conversion function needs the Read instance (see below) to be compatible with the output of a Show instance from an old version of Cabal. If you change the data structure as you've done here, you break the Read instance!